src/api/app/controllers/staging_project/staged_requests_controller.rb

```diff
@@ -2,4 +2,53 @@ class StagingProject::StagedRequestsController < StagingProjectController
   def index
     @requests = @project.staged_requests
   end
+
+  # accepts a list of bs_request numbers e.g. 
+  # tosc api -X POST --data "<requests><number>1</number><number>2</number></requests>" '/staging_project/home:Admin:Staging:A/staged_requests'
```

Comment:
> I get this right that these are only adding requests?

Yes, for now you would need to send a list of request numbers you would like to add for a staging request

```
 tosc api -X POST 
--data "<requests><number>1</number><number>2</number></requests>"
'/staging_project/home:Admin:Staging:A/staged_requests'
```

would add request with number 1 & 2 to home:Admin:Staging:A (branch them & set the association).

We know that the osc plugin also supports e.g. adding all requests by regex but that would happen next / or would stay in osc and several requests are necessary (e.g. first query the numbers and then assigning the numbers). Needs to be discussed.

Reply:
That's not my concern. I'm just wondering if we have to lock requests to avoid 2 staging managers picking different letters. If it's quickly done and I don't need to know the full set, it's not a big problem

Reply:
Hm, we could set the database association in the action and creating the source containers in a background job. Setting the associations should be atomic and afterwards it is not possible to assign them to a different staging project anymore. If something goes wrong in the background job while creating the source containers we could simply release the associations again.

> and I don't need to know the full set

What should that mean?

Reply:
> > and I don't need to know the full set
> 
> What should that mean?

I mean like adding repositories through _meta PUTs where I have to know all repositories - this is very race. If adding means adding numbers instead of repeating all selected requests, it's less racy